### PR TITLE
{go,bats}: `dolt add .` for dolt_docs, with updated `dolt status` and bats tests

### PR DESCRIPTION
**Pull request by katiemcculloch**

### CODE REVIEW

File: go/libraries/doltcore/doltdb/root_val.go [hunk 1]
```diff
@@ -484,50 +492,154 @@ func (root *RootValue) RemoveTables(ctx context.Context, tables ...string) (*Roo
...
 
-	var rowMap types.Map
-	if found {
-		rowMap, err = docTable.GetRowData(ctx)
+	if newTblFound {
+		newRows, err := newTbl.GetRowData(ctx)
 		if err != nil {
 			return nil, nil, nil, err
 		}
+		err = newRows.IterAll(ctx, func(key, val types.Value) error {
```
Comment (katiemcculloch) [inline go/libraries/doltcore/doltdb/root_val.go]: @bheni Lmk if you'd recommend a better way of doing this.... right now I'm iterating through all the newRows (to find modified and appended) and then all the oldRows (to find deleted). I thought about nesting the iterators but didn't want to increase the complexity.

Comment (bheni) [reply go/libraries/doltcore/doltdb/root_val.go]: I think the approach is fine.  But this function has gotten a bit large, and hard to understand what is going on.  Would be clearer if you broke it up into functions with names that clearly describe each stage.

File: go/libraries/doltcore/env/environment.go [hunk 2]
```diff
@@ -650,3 +653,114 @@ func (dEnv *DoltEnv) GetUserHomeDir() (string, error) {
...
+		return err
+	}
+
+	docTbl, found, err := root.GetTable(ctx, doltdb.DocTableName)
+
+	if err != nil {
+		return err
+	}
+
+	if found {
```
Comment (bheni) [inline go/libraries/doltcore/env/environment.go]: Maybe move the code inside this block to a function called something like updateExistingDocs

File: go/libraries/doltcore/env/environment.go [hunk 3]
```diff
@@ -650,3 +653,114 @@ func (dEnv *DoltEnv) GetUserHomeDir() (string, error) {
...
+			}
+			newRoot, err = root.PutTable(ctx, doltdb.DocTableName, docTbl)
+			if err != nil {
+				return err
+			}
+		}
+		return dEnv.UpdateWorkingRoot(ctx, newRoot)
+	}
+
+	// Create a dolt_docs table if it doesn't exist yet
```
Comment (bheni) [inline go/libraries/doltcore/env/environment.go]: move to a function called createDocsTable

File: go/libraries/doltcore/doltdb/root_val.go (hunk 1 above)
Comment (katiemcculloch) [reply go/libraries/doltcore/doltdb/root_val.go]: I broke up the root.DocDiff function quite a bit. I could still probably extract one more helper function from within these iterators. I attempted it today, but got stuck and needed to move on.

File: bats/dolt-docs.bats [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+    echo testing123 > LICENSE.md
+    run cat LICENSE.md
+    [ "$output" = "testing123" ]
+    run dolt add .
+    [ "$status" -eq 0 ]
+    run dolt status
+    [ "$status" -eq 0 ]
+    [[ "$output" =~ "On branch master" ]]
+    [[ "$output" =~ "Changes to be committed:" ]]
+    [[ "${lines[3]}" =~ ([[:space:]]*new doc:[[:space:]]*LICENSE.md) ]] || false
```
[the remark this answers is not in the document]
Comment (katiemcculloch) [reply bats/dolt-docs.bats]: 👍

File: bats/dolt-docs.bats [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+    setup_common
+}
+
+teardown() {
+    teardown_common
+}
+
+@test "dolt status and ls to view docs on dolt init" {
+    run ls
+    [ "${lines[0]}" = "LICENSE.md" ]
```
[the remark this answers is not in the document]
Comment (katiemcculloch) [reply bats/dolt-docs.bats]: 👍

File: go/libraries/doltcore/doltdb/root_val.go [hunk 6]
```diff
@@ -484,50 +492,154 @@ func (root *RootValue) RemoveTables(ctx context.Context, tables ...string) (*Roo
...
+			newRow, newRowFound, err := newTbl.GetRow(ctx, key.(types.Tuple), newSch)
```
[the remark this answers is not in the document]
Comment (katiemcculloch) [reply go/libraries/doltcore/doltdb/root_val.go]: Ooooo thanks!

File: go/libraries/doltcore/env/environment.go (hunk 2 above)
Comment (katiemcculloch) [reply go/libraries/doltcore/env/environment.go]: done!

File: go/libraries/doltcore/env/environment.go (hunk 3 above)
Comment (katiemcculloch) [reply go/libraries/doltcore/env/environment.go]: done!

**Comment by katiemcculloch** [comment]
@bheni this is ready for review again!

**Pull request merged**